go/vt/schemadiff/schema.go

```diff
@@ -134,6 +135,8 @@ func getViewDependentTableNames(createView *sqlparser.CreateView) (names []strin
 // normalize is called as part of Schema creation process. The user may only get a hold of normalized schema.
 // It validates some cross-entity constraints, and orders entity based on dependencies (e.g. tables, views that read from tables, 2nd level views, etc.)
 func (s *Schema) normalize() error {
+	s.named = map[string]Entity{}
+	s.sorted = []Entity{}
```

Comment:
Are these added for a specific reason?

Reply:
They are. At the end of `apply()` we call normalize, and we need to make sure that existing `named` values are erased. `sorted` happens to be unpopulated; but `normalize` really expects both to be empty. 
We could clear both in `apply`, as right now this is the only path where the values get populated before `normalize`. But I feel `normalize` is the safer place to do that.